### PR TITLE
Add filament change button

### PR DESCRIPTION
**Pull request by android444**

### CODE REVIEW

**Comment by thinkyhead** [comment]
In C++ the `if` keyword requires parentheses around the conditions.

File: Marlin/src/MarlinCore.cpp [hunk 1]
```diff
@@ -533,6 +533,20 @@ inline void manage_inactivity(const bool ignore_stepper_queue=false) {
     }
   #endif
 
+  #if HAS_FILAMENTCHANGE
+    // Handle a standalone FILAMENTCHANGE button
+    constexpr millis_t FILAMENTCHANGE_DEBOUNCE_DELAY = 1000UL;
+    static millis_t next_filamentchange_key_ms; // = 0
+    if !READ(FILAMENTCHANGE_PIN)) { // FILAMENTCHANGE_PIN goes LOW when pressed
```
Comment (thinkyhead) [inline Marlin/src/MarlinCore.cpp]: In C++ the if keyword requires parentheses around the conditions.

File: Marlin/src/MarlinCore.cpp [hunk 2]
```diff
@@ -533,6 +533,20 @@ inline void manage_inactivity(const bool ignore_stepper_queue=false) {
     }
   #endif
 
+  #if HAS_FILAMENTCHANGE
```
Comment (thinkyhead) [inline Marlin/src/MarlinCore.cpp:536]: `HAS_FILAMENTCHANGE` is not defined.

As a general principle, please complete and test your feature before submitting the PR.
This saves us the time required to type comments like this one.

File: Marlin/src/MarlinCore.cpp (hunk 1 above)
Comment (android444) [reply Marlin/src/MarlinCore.cpp]: parentheses around the conditions

File: Marlin/src/MarlinCore.cpp (hunk 2 above)
Comment (android444) [reply Marlin/src/MarlinCore.cpp:536]: HAS_FILAMENTCHANGE is defined.

I tested it and it works.

**Comment by android444** [comment]
There are mistakes. I will correct it and add it again.

**Pull request closed without merging**